src/compiler/scala/tools/nsc/typechecker/MethodSynthesis.scala

```diff
@@ -125,14 +125,21 @@ trait MethodSynthesis {
     def enterGetterSetter(tree: ValDef): Unit = {
       val fieldSym =
         if (noFieldFor(tree, owner)) NoSymbol
-        else owner.newValue(tree.name append NameTransformer.LOCAL_SUFFIX_STRING, tree.pos, tree.mods.flags & FieldFlags | PrivateLocal)
+        else owner.newValue(tree.name append NameTransformer.LOCAL_SUFFIX_STRING, tree.pos, tree.mods.flags & (FieldFlags | DEFERRED) | PrivateLocal)
 
       val getter = Getter(tree)
       val getterSym = getter.createSym
 
       // only one symbol can have `tree.pos`, the others must focus their position
       // normally the field gets the range position, but if there is none, give it to the getter
-      tree.symbol = fieldSym orElse (getterSym setPos tree.pos)
+      if (fieldSym != NoSymbol) {
+        tree.symbol = fieldSym
+        val fieldValDef: global.ValDef = treeCopy.ValDef(tree, tree.mods.copy(flags = fieldSym.flags, privateWithin = tpnme.EMPTY), fieldSym.name, tree.tpt, tree.rhs)
+        context.unit.transformed(tree) = fieldValDef
```

Comment:
Okay, I'll try that. That comms channel is a necessary evil to avoid double-typechecking. We should probably replace the current `Map[Tree, Tree]` with something more localized (add the typed tree to the untyped tree as a tree or symbol attachment and have the typer pick it up from there.